odoo/models.py

```diff
@@ -4283,24 +4283,26 @@ def _check_company(self, fnames=None):
         if not (regular_fields or property_fields):
             return
 
+        # If company_field is `0`, then record[company_field] will simply reference the current record
```

Comment:
Since you touches this method, maybe in line 4274, rename `property_fields` to `company_dependent_fields` since we don't have `ir_property` anymore and `property_fields` should only mean `fields.Properties`